### PR TITLE
Feature/event provider

### PR DESCRIPTION
**Pull request by dominikmatt**

### CODE REVIEW

File: Dynamic/Collections/Page.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+
```
Comment (wachterjohannes) [inline Dynamic/Collections/Page.php]: missing file header

File: Dynamic/Collections/Page.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * TODO: add interace
+ * TODO: comments
+ */
+class Page implements FormCollectionTitleInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function CollectionTitleProvider() {
```
Comment (wachterjohannes) [inline Dynamic/Collections/Page.php]: format the whole file

File: Dynamic/FormCollectionTitleInterface.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Sulu\Bundle\FormBundle\Dynamic;
+
+/**
+ * Defines the form type implementation.
+ */
+interface FormCollectionTitleInterface {
```
Comment (wachterjohannes) [inline Dynamic/FormCollectionTitleInterface.php:16]: format the whole file

File: Dynamic/FormCollectionTitleInterface.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?php
+
```
Comment (wachterjohannes) [inline Dynamic/FormCollectionTitleInterface.php]: missing header

File: Dynamic/FormCollectionTitlePool.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<?php
+namespace Sulu\Bundle\FormBundle\Dynamic;
+
+/**
+ * Holds the available Form-Collections Types.
+ */
+class FormCollectionTitlePool {
```
Comment (wachterjohannes) [inline Dynamic/FormCollectionTitlePool.php]: format the whole file

File: Dynamic/FormCollectionTitlePool.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
+<?php
+namespace Sulu\Bundle\FormBundle\Dynamic;
```
Comment (wachterjohannes) [inline Dynamic/FormCollectionTitlePool.php]: missing file header

File: Dynamic/FormCollectionTitlePool.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Holds the available Form-Collections Types.
+ */
+class FormCollectionTitlePool {
+    private $types;
+
+    /**
+     * FormCollectionTitlePool constructor.
```
Comment (wachterjohannes) [inline Dynamic/FormCollectionTitlePool.php]:
```
/**
 * @param array $types
 */
```

File: Dynamic/FormCollectionTitlePool.php (hunk 5 above)
Comment (wachterjohannes) [reply Dynamic/FormCollectionTitlePool.php]: missing interface

File: Dynamic/FormCollectionTitlePool.php [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * Returns all collection-types.
+     *
+     * @return FormCollectionTitleInterface[]
+     */
+    public function all()
+    {
+        return $this->types;
+    }
+}
```
Comment (wachterjohannes) [inline Dynamic/FormCollectionTitlePool.php]: add empty line at file end

File: Dynamic/FormCollectionTitleInterface.php [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+/**
+ * Defines the form type implementation.
+ */
+interface FormCollectionTitleInterface {
+    /**
+     * TODO: implement it
+     */
+    public function CollectionTitleProvider();
+}
```
Comment (wachterjohannes) [inline Dynamic/FormCollectionTitleInterface.php]: add empty line to file end

File: Entity/Dynamic.php [hunk 10]
```diff
@@ -371,18 +376,35 @@ public function getFieldsByType($type)
...
     {
         if (!$this->form) {
             return;
         }
 
         return $this->form->getFieldType($key);
     }
+
+    /**
+     * @return string
```
Comment (wachterjohannes) [inline Entity/Dynamic.php]: missing phpdoc description

File: Entity/Dynamic.php [hunk 11]
```diff
@@ -371,18 +376,35 @@ public function getFieldsByType($type)
...
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * @param string $type
```
Comment (wachterjohannes) [inline Entity/Dynamic.php]: missing phpdoc description

File: Form/Builder.php [hunk 12]
```diff
@@ -61,21 +67,26 @@ class Builder implements BuilderInterface
      *
      * @param RequestStack $requestStack
      * @param FormFieldTypePool $formFieldTypePool
+     * @param FormCollectionTitlePool $formCollectionTitlePool
      * @param FormRepository $formRepository
      * @param CollectionStrategyInterface $collectionStrategy
      * @param FormFactory $formFactory
      * @param string $defaultStructureView
+     *
```
Comment (wachterjohannes) [inline Form/Builder.php]: remove empty line

File: Resources/config/collections.xml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" ?>
+
```
Comment (wachterjohannes) [inline Resources/config/collections.xml]: remove empty line

**Review by wachterjohannes — CHANGES_REQUESTED** [review]

File: Dynamic/Collections/Page.php (hunk 2 above)
Comment (wachterjohannes) [reply Dynamic/Collections/Page.php]: whats that for a function?

Comment (dominikmatt) [reply Dynamic/Collections/Page.php]: I have renamed this method to the original name and updated the todo description.

**Pull request merged**